omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

```diff
@@ -1677,6 +1676,232 @@ def draw_scalebar(self, panel, region_width, page):
+            if c["active"]:
+                channel = c
+                break
+        if not channel:
+            return
+
+        color_ramp = self.get_color_ramp(channel)
+
+        gap = colorbar["gap"]
+        thickness = colorbar["thickness"]
+        ramp_d = {  # Dict of color ramp properties to pass to paste_image
+            'zoom': '100',
+            'dx': 0,
+            'dy': 0,
+            'orig_height': panel['orig_height'],
+            'orig_width': panel['orig_width']
+        }
+        start, end = channel["window"]["start"], channel["window"]["end"]
+
+        decimals = max(0, int(numpy.ceil(
+            -numpy.log10((end - start) / colorbar["num_ticks"]))))
+        labels = numpy.linspace(start, end, num=colorbar["num_ticks"])
+        # Round the label str to the appropriate decimal
+        labels = [f"{label:.{decimals}f}" for label in labels]
+
+        pos_ratio = numpy.linspace(0, 1, colorbar["num_ticks"])
+        if colorbar["position"] in ["left", "right"]:
+            color_ramp = color_ramp.transpose((1, 0, 2))[::-1]
+            ramp_d['width'] = thickness
+            ramp_d['height'] = panel['height']
+            ramp_d['y'] = panel['y']
+            ramp_d['x'] = panel['x'] - (gap + thickness)
+            labels_x = [ramp_d['x']]
+            labels_y = ramp_d['y'] + panel['height'] * pos_ratio
+            labels = labels[::-1]
+            ramp_d["align"] = "right"
+            if colorbar["position"] == "right":
+                ramp_d['x'] = panel['x'] + panel['width'] + gap
+                labels_x = [ramp_d['x'] + ramp_d['width']]
+                ramp_d["align"] = "left"
+            labels_x *= len(labels)  # Duplicate x postions
+        elif colorbar["position"] in ["top", "bottom"]:
+            ramp_d['width'] = panel['width']
+            ramp_d['height'] = thickness
+            ramp_d['x'] = panel['x']
+            ramp_d['y'] = panel['y'] - (gap + thickness)
+            labels_x = ramp_d['x'] + panel['width'] * pos_ratio
+            labels_y = [ramp_d['y']]
+            ramp_d["align"] = "center"
+            if colorbar["position"] == "bottom":
+                ramp_d['y'] = panel['y'] + panel['height'] + gap
+                labels_y = [ramp_d['y'] + ramp_d['height']]
+            labels_y *= len(labels)  # Duplicate y postions
+
+        pil_img = Image.fromarray(color_ramp)
+        img_name = channel["color"] + ".png"
+
+        # for PDF export, we might have a target dpi
+        dpi = panel.get('min_export_dpi', None)
+
+        # Paste the panel to PDF or TIFF image
+        self.paste_image(pil_img, img_name, ramp_d, page, dpi,
+                         is_colorbar=True)
+
+        # Handle page offsets
+        ramp_d['x'] -= page['x']
+        ramp_d['y'] -= page['y']
+        labels_x = numpy.array(labels_x) - page['x']
+        labels_y = numpy.array(labels_y) - page['y']
+
+        self.draw_colorbar_ticks(colorbar, ramp_d, labels, labels_x, labels_y)
```

Comment:
that's where I had to split the code between PDF and TIFF export. With the TIFF, I need extra code to handle pixel offsets.